datafusion/core/src/datasource/listing/table.rs

```diff
@@ -1430,15 +1476,18 @@ mod tests {
         table_prefix: &str,
         target_partitions: usize,
         output_partitioning: usize,
+        file_ext: Option<&str>,
     ) -> Result<()> {
         let ctx = SessionContext::new();
         register_test_store(&ctx, &files.iter().map(|f| (*f, 10)).collect::<Vec<_>>());
 
         let format = AvroFormat {};
 
-        let opt = ListingOptions::new(Arc::new(format))
-            .with_file_extension("")
+        let mut opt = ListingOptions::new(Arc::new(format))
```

Comment:
Maybe we could make this a nicer API by adding `with_file_extension_opt` or something so this code could be like:

```rust
let opt = ListingOptions::new(Arc::new(format))
            .with_file_extension_opt(file_ext)
            .with_target_partitions(target_partitions);
```

Reply:
Here is a PR with the proposed API: https://github.com/apache/datafusion/pull/12461

Reply:
> Here is a PR with the proposed API: #12461

I have modified the code according to this API.